lib/ui/window.dart

```diff
@@ -92,6 +92,18 @@ class FrameTiming {
   FrameTiming(List<int> timestamps)
       : assert(timestamps.length == FramePhase.values.length), _timestamps = timestamps;
 
+  /// Construct [FrameTiming] with given timestamp in micrseconds.
+  ///
+  /// TODO(CareF): This is part of #20229. Remove back to default constructor
+  /// after #20229 lands and corresponding framwork PRs land.
+  FrameTiming.fromTimeStamps({
+    int? vsyncStart,
+    required int buildStart,
+    required int buildFinish,
+    required int rasterStart,
+    required int rasterFinish
+  }) : _timestamps = <int>[buildStart, buildFinish, rasterStart, rasterFinish];
```

Comment:
Nit: make this `factory FrameTiming.fromTimeStamps` so it can use `FrameTiming(List<int>)` inside the implementation for a length check.

We should also document that this constructor is meant for unit tests only. Real `FrameTiming`s should be retrieved from `Window.onReportTimings`.

Reply:
Done